skimage/transform/_geometric.py

```diff
@@ -1148,40 +1109,15 @@ def __call__(self, coords):
 
         return out
 
-    def inverse(self, coords):
-        """Apply inverse transformation.
-
-        Coordinates outside of the mesh will be set to `- 1`.
-
-        Parameters
-        ----------
-        coords : (N, D) array_like
-            Source coordinates.
-
-        Returns
-        -------
-        coords : (N, D) array
-            Transformed coordinates.
-
-        """
-        coords = np.asarray(coords)
-        out = np.empty_like(coords, np.float64)
-
-        # determine triangle index for each coordinate
-        simplex = self._inverse_tesselation.find_simplex(coords)
-
-        # coordinates outside of mesh
-        out[simplex == -1, :] = -1
-
-        for index in range(len(self._inverse_tesselation.simplices)):
-            # affine transform for triangle
-            affine = self.inverse_affines[index]
-            # all coordinates within triangle
-            index_mask = simplex == index
-
-            out[index_mask, :] = affine(coords[index_mask, :])
-
-        return out
+    @property
+    def inverse(self):
+        """Return a transform object representing the inverse."""
+        tform = type(self)()
```

Comment:
Maybe `PiecewiseAffineTransform()`?

Reply:
I saw this proposed in the [issue](https://github.com/scikit-image/scikit-image/issues/6921#issuecomment-1542312019). I can also use explicit classes for the others as well, if preferred?

Reply:
Not a big deal. I'd use `type` for general applications, and the specific class for the case where you know what the class name is up-front.

Reply:
Actually, I think `type(self)` would be the better practice here. It will allow an inheriting classes X to use this property and return a new inverse of X (the same class). Do we want to explicitly block that behavior by using an explicit class here?

Reply:
That reasoning makes sense, I learned something! I reverted it back to `type(self)`.